aui.views/src/AUI/Platform/linux/ADBus.cpp

```diff
@@ -111,28 +111,76 @@ DBusHandlerResult ADBus::listener(DBusConnection     *connection,
+        aui::dbus::iter_append<bool>(&sub, std::get<bool>(t));
+    } else if (std::holds_alternative<std::int16_t>(t)) {
+        dbus_message_iter_open_container(iter, DBUS_TYPE_VARIANT, converter<std::int16_t>::signature.c_str(), &sub);
+        aui::dbus::iter_append<std::int16_t>(&sub, std::get<std::int16_t>(t));
+    } else if (std::holds_alternative<std::uint16_t>(t)) {
+        dbus_message_iter_open_container(iter, DBUS_TYPE_VARIANT, converter<std::uint16_t>::signature.c_str(), &sub);
+        aui::dbus::iter_append<std::uint16_t>(&sub, std::get<std::uint16_t>(t));
+    } else if (std::holds_alternative<std::int32_t>(t)) {
+        dbus_message_iter_open_container(iter, DBUS_TYPE_VARIANT, converter<std::int32_t>::signature.c_str(), &sub);
+        aui::dbus::iter_append<std::int32_t>(&sub, std::get<std::int32_t>(t));
+    } else if (std::holds_alternative<std::uint32_t>(t)) {
+        dbus_message_iter_open_container(iter, DBUS_TYPE_VARIANT, converter<std::uint32_t>::signature.c_str(), &sub);
+        aui::dbus::iter_append<std::uint32_t>(&sub, std::get<std::uint32_t>(t));
+    } else if (std::holds_alternative<std::int64_t>(t)) {
+        dbus_message_iter_open_container(iter, DBUS_TYPE_VARIANT, converter<std::int64_t>::signature.c_str(), &sub);
+        aui::dbus::iter_append<std::int64_t>(&sub, std::get<std::int64_t>(t));
+    } else if (std::holds_alternative<std::uint64_t>(t)) {
+        dbus_message_iter_open_container(iter, DBUS_TYPE_VARIANT, converter<std::uint64_t>::signature.c_str(), &sub);
+        aui::dbus::iter_append<std::uint64_t>(&sub, std::get<std::uint64_t>(t));
+    } else if (std::holds_alternative<double>(t)) {
+        dbus_message_iter_open_container(iter, DBUS_TYPE_VARIANT, converter<double>::signature.c_str(), &sub);
+        aui::dbus::iter_append<double>(&sub, std::get<double>(t));
+    } else if (std::holds_alternative<std::string>(t)) {
+        dbus_message_iter_open_container(iter, DBUS_TYPE_VARIANT, converter<std::string>::signature.c_str(), &sub);
+        aui::dbus::iter_append<std::string>(&sub, std::get<std::string>(t));
+    } else if (std::holds_alternative<ObjectPath>(t)) {
+        dbus_message_iter_open_container(iter, DBUS_TYPE_VARIANT, converter<ObjectPath>::signature.c_str(), &sub);
+        aui::dbus::iter_append<ObjectPath>(&sub, std::get<ObjectPath>(t));
+    } else if (std::holds_alternative<AVector<std::uint8_t>>(t)) {
+        dbus_message_iter_open_container(iter, DBUS_TYPE_VARIANT, converter<AVector<std::uint8_t>>::signature.c_str(), &sub);
+        aui::dbus::iter_append<AVector<std::uint8_t>>(&sub, std::get<AVector<std::uint8_t>>(t));
+    } else {
+        dbus_message_iter_open_container(iter, DBUS_TYPE_VARIANT, converter<std::nullopt_t>::signature.c_str(), &sub);
+        aui::dbus::iter_append<std::nullopt_t>(&sub, std::nullopt);
+    }
+
+    dbus_message_iter_close_container(iter, &sub);
```

Comment:
Let's make simpler:
```suggestion
    std::visit([&]<typename T>(const T& containedValue) {
      DBusMessageIter sub;
      dbus_message_iter_open_container(iter, DBUS_TYPE_VARIANT, converter<T>::signature.c_str(), &sub);
      AUI_DEFER { dbus_message_iter_close_container(iter, &sub); };
      aui::dbus::iter_append<T>(&sub, containedValue);
    }, t);
```

Reply:
Fixed, thanks! I was not even close to thinking like this haha.